app/models/article.rb

```diff
@@ -65,6 +69,10 @@ def self.delete_orphan_articles
   def related_article_ranks
 
   end
+  
+  def gis_enabled?
+    self.categories.any? { |c| c.gis_enabled }
```

Comment:
Can we rewrite this to push the logic into the database query, rather than needing to loop through results?  `self.categories.where(:gis_enabled => true).present?` might work.